aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -116,46 +190,70 @@ Tensor mean(const Tensor &self) {
   return at::native::mean(self, nullopt);
 }
 
-static inline Tensor sum(const Tensor &self, optional<ScalarType> dtype) {
-  return at::_sum(integer_upcast(self, dtype));
+static ScalarType get_dtype(Tensor& result, const Tensor& self, optional<ScalarType> dtype,
+                            bool promote_integers=false) {
+  if (dtype.has_value()) {
+    return dtype.value();
```

Comment:
this is a change of behavior, right?  Previously we checked that if dtype.has_value then dtype.value() == result.type().scalarType().  Why the change?